include/deal.II/lac/full_matrix.h

```diff
@@ -65,12 +65,9 @@ class FullMatrix : public Table<2,number>
 {
 public:
   /**
-   * A type of used to index into this container. Because we can not expect to
-   * store matrices bigger than what can be indexed by a regular unsigned
-   * integer, <code>unsigned int</code> is completely sufficient as an index
-   * type.
+   * A type used to index into this container.
    */
-  typedef unsigned int size_type;
+  typedef std::size_t size_type;
```

Comment:
What's wrong with the previous explanation of why we use `unsigned int`? Or are you thinking of matrices of size 1B x 1?

Reply:
Since we want `std::size_t` here, I consider changing the text to 
```
A type of used to index into this container. Because we can not expect to
store matrices bigger than what can be indexed by a std::size_t, 
<code>std::size_t</code> is completely sufficient as an index type.
```
more confusing than helpful.

>  Or are you thinking of matrices of size 1B x 1?

Yes, that is what you were intending in https://github.com/dealii/dealii/pull/5652#issuecomment-353243464.

Reply:
I see. Yes, then the reduced text seems reasonable. (The old text was also badly broken with regards to grammar -- "A type of...".)